src/components/system/SystemAlert.jsx

```diff
@@ -0,0 +1,130 @@
+import React, { Component } from "react";
+import { connect } from 'react-redux';
+import PropTypes from "prop-types";
+import * as R from 'ramda';
+import moment from 'moment';
+import { differenceInMinutes, differenceInHours } from 'date-fns';
+import systemActions from '../../features/system/actions';
+import '../../../assets/css/widgets.css';
+
+class Alert extends Component {
+
+  componentDidMount() {
+    this.getSystemInfo();
+  }
+
+  componentDidUpdate() {
+    this.getSystemInfo();
+  }
+
+  getSystemInfo() {
+    const { session, dispatch } = this.props;
+
+    if (!R.isEmpty(session) && session.authenticated) {
+      dispatch(systemActions.systemPollStartAction());
+    } else {
+      dispatch(systemActions.systemPollStopAction());
+    }
+  }
+
+  renderAlertMessage() {
+    const { system } = this.props;
+    const deviceTime = new Date();
+    const serverTime = new Date(`${system.systemDate}T${system.systemTime}Z`);
+
+    const deviceTimeInUTC = deviceTime.toISOString();
+
+    const hasTimeZoneOffset = serverTime.getTimezoneOffset() === deviceTime.getTimezoneOffset();
+    const timeOffset = Math.abs(
+      differenceInHours(
+        new Date(`${system.systemDate}T${system.systemTime}`),
+        deviceTime
+      )
+    );
+    const timeDifference = Math.abs(
+      differenceInMinutes(
+        serverTime,
+        deviceTimeInUTC
+      ));
+
+    const hasTimeDifference = timeDifference > 5;
+    const hasConnection = system.systemConnection || navigator.onLine;
```

Comment:
I think this should just rely on system.systemConnection or should be an &&... "navigator.onLine" may not be sensitive enough... ie, it can return TRUE during times when the server is actually offline